devstack/devstack.dockerfile

```diff
@@ -0,0 +1,68 @@
+#
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y git sudo python python3-dev python3 python3-venv ca-certificates unzip openssh-server mongodb mysql-server mysql-client unzip gcc make
+#
+# Install firefox 52 because it is selenium < 3 compatible
+#
+RUN apt-get install -y software-properties-common && add-apt-repository --yes ppa:jonathonf/firefox-esr-52 && apt-get update && apt-get install -y firefox-esr
+#
+# Install postgresql 10 because circle.yml has that version mentionned, not sure it would make a difference to keep 9
+# 
+RUN echo 'deb http://apt.postgresql.org/pub/repos/apt/ xenial-pgdg main' >> /etc/apt/sources.list.d/pgdg.list && wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y postgresql-10
+#
+# All commands are run via exec under a non privileged user that has the same name as the user building the image
+# and the same id. So that volumes mounted from a directory they own are not unexpectedly chowned to root in various
+# places.
+#
+RUN if test $USER_NAME != root ; then useradd --no-create-home --home-dir /tmp --uid $USER_ID $USER_NAME && echo "$USER_NAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers ; fi
+#
+# en_US.UTF-8 is assumed to always be installed
+#
+RUN apt-get install -y locales && locale-gen en_US.UTF-8
+ENV LANG=en_US.UTF-8
+#
+# Install consul (should be in a makefile target common to circle.yml and devstack)
+#
+RUN wget -P /tmp https://releases.hashicorp.com/consul/1.2.1/consul_1.2.1_linux_amd64.zip && sudo unzip /tmp/consul_1.2.1_linux_amd64.zip -d /usr/local/bin
+COPY devstack/consul.service /etc/systemd/system/consul.service
+RUN systemctl enable consul
+#
+# Install system dependencies (should be the only place where an install happens)
+#
+WORKDIR /opt
+COPY Makefile .
+COPY debian_packages.lst .
+RUN mkdir devstack && touch devstack/Makefile && make install_system_dependencies
+#
+# Setup a virtual environment owned by the user with all python dependencies installed
+#
+COPY requirements.txt .
+COPY cleanup_utils/requirements.txt cleanup_utils/requirements.txt
+RUN python3 -m venv venv && . venv/bin/activate && pip install --upgrade pip && pip install --upgrade virtualenv && pip install -r requirements.txt && pip install -r cleanup_utils/requirements.txt && chown -R $USER_NAME venv
+#
+# Install a one shot script to be run when systemd is finished spawning services
+# and carry out all actions that cannot be run when building the dockerfile, such
+# as creating database users.
+#
+COPY devstack/setup-devstack.sh /usr/local/bin/setup-devstack.sh
+RUN sed -i -e "s/%%USER_NAME%%/$USER_NAME/g" \
+           -e "s/%%USER_ID%%/$USER_ID/g" \
+	   /usr/local/bin/setup-devstack.sh
+COPY devstack/setup-devstack.service /etc/systemd/system/setup-devstack.service
+RUN systemctl enable setup-devstack
```

Comment:
If this is a one-shot command, why not run it while provisioning the devstack? 
Eg: When running `make devstack.provision`, create dbs, users, apply migrations and leave the server ready to run.

Reply:
This goes back to what I think is a misunderstanding on the scope of this PR: it does not deploy Ocim.